src/graphic/Fast3D/gfx_pc.cpp

```diff
@@ -2578,670 +2620,1067 @@ void gfx_set_framebuffer(int fb, float noise_scale);
+    Gfx* cmd = *cmd0;
+
+    gfx_flush();
+    gfx_rapi->select_texture_fb((uint32_t)cmd->words.w1);
+    g_rdp.textures_changed[0] = false;
+    g_rdp.textures_changed[1] = false;
+    return false;
+}
+
+// TODO handle special OTR opcodes later...
+bool gfx_set_grayscale_handler_custom(Gfx** cmd0) {
+    Gfx* cmd = *cmd0;
+
+    g_rdp.grayscale = cmd->words.w1;
+    return false;
+}
+
+bool gfx_load_block_handler_rdp(Gfx** cmd0) {
+    Gfx* cmd = *cmd0;
+
+    gfx_dp_load_block(C1(24, 3), C0(12, 12), C0(0, 12), C1(12, 12), C1(0, 12));
+    return false;
+}
+
+bool gfx_load_tile_handler_rdp(Gfx** cmd0) {
+    Gfx* cmd = *cmd0;
+
+    gfx_dp_load_tile(C1(24, 3), C0(12, 12), C0(0, 12), C1(12, 12), C1(0, 12));
+    return false;
+}
+
+bool gfx_set_tile_handler_rdp(Gfx** cmd0) {
+    Gfx* cmd = *cmd0;
+
+    gfx_dp_set_tile(C0(21, 3), C0(19, 2), C0(9, 9), C0(0, 9), C1(24, 3), C1(20, 4), C1(18, 2), C1(14, 4), C1(10, 4),
+                    C1(8, 2), C1(4, 4), C1(0, 4));
+    return false;
+}
+
+bool gfx_set_tile_size_handler_rdp(Gfx** cmd0) {
+    Gfx* cmd = *cmd0;
+
+    gfx_dp_set_tile_size(C1(24, 3), C0(12, 12), C0(0, 12), C1(12, 12), C1(0, 12));
+    return false;
+}
+
+bool gfx_load_tlut_handler_rdp(Gfx** cmd0) {
+    Gfx* cmd = *cmd0;
+
+    gfx_dp_load_tlut(C1(24, 3), C1(14, 10));
+    return false;
+}
+
+bool gfx_set_env_color_handler_rdp(Gfx** cmd0) {
+    Gfx* cmd = *cmd0;
+
+    gfx_dp_set_env_color(C1(24, 8), C1(16, 8), C1(8, 8), C1(0, 8));
+    return false;
+}
+
+bool gfx_set_prim_color_handler_rdp(Gfx** cmd0) {
+    Gfx* cmd = *cmd0;
+
+    gfx_dp_set_prim_color(C0(8, 8), C0(0, 8), C1(24, 8), C1(16, 8), C1(8, 8), C1(0, 8));
+    return false;
+}
+
+bool gfx_set_fog_color_handler_rdp(Gfx** cmd0) {
+    Gfx* cmd = *cmd0;
+
+    gfx_dp_set_fog_color(C1(24, 8), C1(16, 8), C1(8, 8), C1(0, 8));
+    return false;
+}
+
+bool gfx_set_blend_color_handler_rdp(Gfx** cmd0) {
+    Gfx* cmd = *cmd0;
+
+    gfx_dp_set_blend_color(C1(24, 8), C1(16, 8), C1(8, 8), C1(0, 8));
+    return false;
+}
+
+bool gfx_set_fill_color_handler_rdp(Gfx** cmd0) {
+    Gfx* cmd = *cmd0;
+
+    gfx_dp_set_fill_color((uint32_t)cmd->words.w1);
+    return false;
+}
+
+bool gfx_set_intensity_handler_custom(Gfx** cmd0) {
+    Gfx* cmd = *cmd0;
+
+    gfx_dp_set_grayscale_color(C1(24, 8), C1(16, 8), C1(8, 8), C1(0, 8));
+    return false;
+}
+
+bool gfx_set_combine_handler_rdp(Gfx** cmd0) {
+    Gfx* cmd = *cmd0;
+
+    gfx_dp_set_combine_mode(
+        color_comb(C0(20, 4), C1(28, 4), C0(15, 5), C1(15, 3)), alpha_comb(C0(12, 3), C1(12, 3), C0(9, 3), C1(9, 3)),
+        color_comb(C0(5, 4), C1(24, 4), C0(0, 5), C1(6, 3)), alpha_comb(C1(21, 3), C1(3, 3), C1(18, 3), C1(0, 3)));
+    return false;
+}
+
+bool gfx_tex_rect_and_flip_handler_rdp(Gfx** cmd0) {
+    Gfx* cmd = *cmd0;
+    uint32_t opcode = (uint32_t)(cmd->words.w0 >> 24);
+    int32_t lrx, lry, tile, ulx, uly;
+    uint32_t uls, ult, dsdx, dtdy;
 #ifdef F3DEX_GBI_2E
```

Comment:
do we still want these ifdefs in the functions or do we want to have different functions for each and ifdef somewhere else?